src/react-components/tokens/CreateTokenContainer.js

```diff
@@ -0,0 +1,169 @@
+      return { ...state, scopes: action.scopes };
+    case CreateTokenActions.fetchingScopesError:
+      return { ...state, error: "Error fetching scopes, please try again later." };
+    case CreateTokenActions.showNoScopesError:
+      return { ...state, showNoScopesSelectedError: true };
+    case CreateTokenActions.toggleScopeChange: {
+      return {
+        ...state,
+        selectedScopes: state.selectedScopes.includes(action.scopeName)
+          ? state.selectedScopes.filter(name => name !== action.scopeName)
+          : [...state.selectedScopes, action.scopeName]
+      };
+    }
+    case CreateTokenActions.toggleTokenTypeChange: {
+      return {
+        ...state,
+        selectedTokenType: action.tokenTypeValue
+      };
+    }
+    default:
+      return state;
+  }
+}
+
+function useCreateToken() {
+  const [state, dispatch] = useReducer(createTokenReducer, initialCreateTokenState);
+
+  const onCreateToken = async ({ tokenType, scopes }) => {
+    if (scopes.length === 0) return dispatch({ type: CreateTokenActions.showNoScopesError });
+
+    dispatch({ type: CreateTokenActions.submitCreateToken });
+
+    try {
+      const tokenInfoObj = await createToken({ tokenType, scopes });
+      const token = tokenInfoObj.credentials[0].token;
+      dispatch({ type: CreateTokenActions.createTokenSuccess, token });
+    } catch (err) {
+      dispatch({ type: CreateTokenActions.createTokenError, errorMsg: err.message });
+    }
+  };
+
+  const fetchScopes = async () => {
+    try {
+      const fetchedScopes = await fetchAvailableScopes();
+      const scopes = fetchedScopes.scopes;
+      dispatch({ type: CreateTokenActions.fetchingScopesSuccess, scopes });
+    } catch (err) {
+      dispatch({ type: CreateTokenActions.fetchingScopesError, errorMsg: err.message });
+    }
+  };
+
+  const toggleSelectedScopes = scopeName => {
+    dispatch({ type: CreateTokenActions.toggleScopeChange, scopeName });
+  };
+
+  const toggleTokenType = tokenTypeValue => {
+    dispatch({ type: CreateTokenActions.toggleTokenTypeChange, tokenTypeValue });
+  };
+
+  return {
+    isPending: state.isPending,
+    scopes: state.scopes,
+    selectedScopes: state.selectedScopes,
+    token: state.token,
+    error: state.error,
+    showNoScopesSelectedError: state.showNoScopesSelectedError,
+    onCreateToken,
+    fetchScopes,
+    toggleSelectedScopes,
+    toggleTokenType,
+    selectedTokenType: state.selectedTokenType,
+    showRevealTokenModal: state.showRevealTokenModal
+  };
+}
+
+export const CreateTokenContainer = ({ onClose }) => {
+  const {
+    scopes,
+    selectedScopes,
+    showRevealTokenModal,
+    token,
+    error,
+    showNoScopesSelectedError,
+    onCreateToken,
+    fetchScopes,
+    toggleSelectedScopes,
+    toggleTokenType,
+    selectedTokenType,
+    isPending
+  } = useCreateToken();
+
+  useEffect(
+    () => {
+      fetchScopes();
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I fix the below line and fetch scopes call infinitely runs. This is the only way I've found that doesn't create an infinite update loop.